packages/algob/src/lib/files.ts

```diff
@@ -64,3 +72,27 @@ export function loadFromYamlFileSilentWithMessage (
     return defaultYamlValue(options);
   }
 }
+
+export async function readMsigFromFile (filename: string): Promise<Object> {
+  if (!filename.endsWith(msigExt)) {
+    throw new Error(`filename "${filename}" must end with "${msigExt}"`);
+  }
+  try {
+    const p = path.join(ASSETS_DIR, filename);
+
+    // Extracting teal code from .mlsig and dumping to .teal to get logic signature
+    const tealFile = filename.split(msigExt)[0] + tealExt;
+    const tealPath = path.join(ASSETS_DIR, tealFile); // assets/<file_name>.teal
+    const tealCode = fs.readFileSync(p, 'utf8').split("LogicSig: ")[0];
+
+    // Write logic code to file with .teal ext
+    writeFile(tealPath, tealCode);
+
+    // return msig object of logic signature to decode further
+    const Msig = fs.readFileSync(p, 'utf8').split("LogicSig: ")[1];
```

Comment:
don't read and split the file twice. you already did it in line 86

Reply:
ok